src/operator/bilinear_sampler.cc

```diff
@@ -28,7 +28,7 @@
 namespace mshadow {
 template<typename DType>
 bool between(DType value, int lowerBound, int upperBound) {
-  return (value >= lowerBound && value <= upperBound);
+  return (static_cast<int>(value) >= lowerBound && static_cast<int>(value) <= upperBound);
```

Comment:
That is not working. We still get WsignCompare warning. Is DType generic to both unsigned and signed types? Besides it is probably better (for values in range of unsigned) to cast signed to unsigned than the other way.